cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1538,6 +1576,14 @@ func generateMountsModeRun(mst *initramfsMountsState) error {
 		}
 	}
 
+	// Setup bind-mounts in /sysroot
+	if err := maybeDoBaseBindMount(); err != nil {
```

Comment:
@anonymouse64 @xnox I'm confused but if we do this only for run mode but to do something it requires the units not to be present, will this not break recover and install, or is the mounting to the tmpfs done by different units in that case?

Reply:
In any UC20 mode (run / install / recover) sysroot.mount is a bind-mount of /run/mnt/ubuntu-data. Whereas in the classic mode (i.e. cloudimg-rootfs) it will be a direct mount of the rootfs.

Also in UC20 mode bind mounts of the kernel snap's modules & firmware must happen in the initrd. Whereas in classic mode there is no kernel snap, and modules/firmware is expected to be available in the rootfs already.

Currently the bind mounts are present as static units in the core-initrd, which make it impossible to make it usable for either UC20 or classic modes simultaniously.

This is similar change, to how in the past "ubuntu-data" was mounted with static units baked into the core-initrd, but then during early UC20 development got moved to be generated by snap-bootstrap. Which has allowed for more strict checks on which ubuntu-data is mounted.

Reply:
Indeed this PR as it stands is insufficient, I am unable to boot a UC20 system in install or recover mode with the mount units removed from the initrd and this branch as-is. It just hangs in the initrd presumably because sysroot-writable has 
`After=run-mnt-base.mount` in it, and when we are running in install mode, `maybeDoBaseBindMount()` doesn't get called and so we are stuck forever in the initrd.

I _think_ it should be enough to add calls to these two functions in `generateMountsCommonInstallRecover()` after line 1277, the `for _, essentialSnap := range essSnaps {` loop, but I haven't successfully tested this yet.

I'm working on a spread test to test the new case where these files are absent from the initrd to see that it passes (I expect it won't ootb with this branch), and then in the future when the initrd is updated to remove these mount units, we can flip the tests to have this spread test _add_ those mount units to the initrd to confirm that it boots properly.